src/data/mllist.lean

```diff
@@ -10,74 +10,126 @@ This isn't so bad, as the typical use is with the tactic monad, in any case.
 
 As we're in meta anyway, we don't bother with proofs about these constructions.
 -/
-
+import data.option.basic
 universes u v
 
 meta inductive mllist (m : Type u → Type u) (α : Type u) : Type u
 | nil {} : mllist
-| cons : α → m mllist → mllist
+| cons : m (option α × mllist) → mllist
+-- | skip : m mllist → mllist
```

Comment:
I think you want something more like this:
```lean
meta mutual inductive mllist_aux, mllist (m : Type u → Type u) (α : Type u)
with mllist_aux : Type u
| nil : mllist_aux
| cons : α → mllist → mllist_aux
with mllist : Type u
| mk : m mllist_aux → mllist
```

Reply:
or 

```lean
meta inductive mllist (m : Type u → Type u) (α : Type u) : Type u
| cons : m (option (α x m mllist)) → mllist
```

Reply:
Oh! I wonder if we can make a recursive type using fixpoints:
```lean
meta def mllist (m : Type u → Type u) (α : Type u) : Type u := m (option (α × mllist))
meta def mllist.nil {m} [monad m] {α} : mllist m α := (return none : m _)
meta def mllist.cons {m} [monad m] {α}
  (a : α) (l : mllist m α) : mllist m α :=
(return (some (a, l)) : m _) -- error: can't defeq
```
too bad...